controller/lib/hal/hal_stm32.h

```diff
@@ -83,6 +70,13 @@ inline void GPIO_OutType(GPIO_Regs *const gpio, int pin, GPIO_OutType outType) {
     gpio->outType &= ~(1 << pin);
 }
 
+enum class GPIO_OutSpeed { LOW = 0, MEDIUM = 1, HIGH = 2, SMOKIN = 3 };
+inline void GPIO_OutSpeed(GPIO_Regs *const gpio, int pin, GPIO_OutSpeed speed) {
+  int S = static_cast<int>(speed);
```

Comment:
Sure.  Although anyone really interested is strongly encouraged to review the register descriptions in the reference manual.

Reply:
It's good to have something close to the code. For straight up constant values it's ok to say "go back and read the manual", but if there is math happening it would be safer to have a copy nearby.